parsing/padding.go

```diff
@@ -0,0 +1,113 @@
+package parsing
+
+import (
+	"errors"
+	"github.com/filecoin-project/go-data-segment/fr32"
+)
+
+// Pad pads a general byte array in to Fr32 chunks of bytes where the topmost bits of the most significant byte are 0
+func Pad(unpaddedData *[]byte) ([]fr32.Fr32, error) {
+	if unpaddedData == nil || len(*unpaddedData) == 0 {
+		return nil, errors.New("empty input")
+	}
+	// Compute amount of Fr32 elements in the result
+	chunkCount := Ceil(len(*unpaddedData)*8, fr32.BitsNeeded)
+	paddedData := make([]fr32.Fr32, chunkCount, chunkCount)
+	bitIdx := 0
+	for i := 0; i < chunkCount; i++ {
+		unpaddedChunk := getChunk(bitIdx, unpaddedData)
+		paddedData[i] = fr32.Fr32{Data: shiftChunk(bitIdx, unpaddedChunk)}
+		// Update bitIdx to the byte we need to start at which is 254 in
+		bitIdx += fr32.BitsNeeded
+	}
+	return paddedData, nil
+}
+
+// Return a slice containing the next segment of unpadded data (without copying data), it will be a slice of either 32 or 33 bytes
+func getChunk(bitIdx int, unpaddedData *[]byte) []byte {
```

Comment:
The pointer to slice snuck through.

Reply:
Thanks! Fixed in d2c13c6e41f87ffbf9e808d71a6271df03a274cc